Snowflake.Data/Logger/EasyLoggerUtil.cs

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System;
+using System.Diagnostics;
+using Mono.Unix;
+
+namespace Snowflake.Data.Log
+{
+    internal class EasyLoggerUtil
+    {
+        internal static int AllPermissions = 777;
+
+        internal static int AllUserPermissions = 700;
+
+        internal static int OnlyUserHasPermissionToWrite = 644;
+
+        internal static int ConvertFileAccessPermissionsToInt(FileAccessPermissions permissions)
+        {
+            int userPermission = 0;
+            int groupPermission = 0;
+            int otherPermission = 0;
+
+            foreach (FileAccessPermissions perm in Enum.GetValues(typeof(FileAccessPermissions)))
+            {
+                // Check if enum exists in the directory permission
+                if ((permissions & perm) != 0)
+                {
+                    string permName = perm.ToString();
+                    if (!permName.Contains("ReadWriteExecute"))
+                    {
+                        if (permName.Contains("User"))
+                        {
+                            userPermission += (int)perm;
+                        }
+                        else if (permName.Contains("Group"))
+                        {
+                            groupPermission += (int)perm;
+                        }
+                        else if (permName.Contains("Other"))
+                        {
+                            otherPermission += (int)perm;
+                        }
+                    }
+                }
+            }
+
+            // Divide by 2^6
+            userPermission /= 64;
+
+            // Divide by 2^3
+            groupPermission /= 8;
+
+            return Convert.ToInt32(string.Format("{0}{1}{2}", userPermission, groupPermission, otherPermission));
+        }
+
+        internal static bool OnlyUserHasAllPermissions(FileAccessPermissions permissions)
+        {
+            return permissions == FileAccessPermissions.UserReadWriteExecute;
+        }
+
+        internal static bool OnlyUserCanWrite(FileAccessPermissions permissions)
+        {
+            return permissions == FileAccessPermissions.UserWrite;
+        }
+
+        internal static string CallBash(string command)
```

Comment:
this is a dangerous stuff. Such things should be available only in tests.

Reply:
I agree, so far I've not been able to find a way to get the umask value without bash. And the `Mono.Unix` package doesn't seem to have a way to get the current umask value either. But the recent changes no longer calls bash in production

Reply:
Fixed